src/pages/account/[address].test.tsx

```diff
@@ -0,0 +1,62 @@
+import { describe } from '@jest/globals';
+import { getAddress } from 'viem';
+
+import AccountProfilePage, { Head } from './[address]';
+import { render, getMockSiteMetadata } from '../../utils/test-utils';
+
+jest.mock('../../features/account/components/AccountInfo', () => ({
+  AccountInfo: () => <div />,
+}));
+
+jest.mock('viem', () => ({
+  getAddress: jest.fn(),
+}));
+
+describe('Account Profile page', () => {
+  let mockGetAddress: jest.Mock;
+
+  beforeEach(() => {
+    mockGetAddress = getAddress as jest.Mock;
+    mockGetAddress.mockClear();
+  });
+
+  it('renders', async () => {
+    const address = '0x6B24aE0ABbeb67058D07b891aF415f288eA57Cc7';
+    mockGetAddress.mockReturnValue(address);
+
+    const { queryByTestId } = render(
+      <AccountProfilePage params={{ address }} />,
+    );
+
+    expect(queryByTestId('account-profile-not-found')).not.toBeInTheDocument();
+    expect(queryByTestId('account-profile')).toBeInTheDocument();
+  });
+
+  it('renders not found page if parameter `address` is incorrect', async () => {
+    mockGetAddress.mockImplementation(() => {
+      throw new Error('Incorrect address');
+    });
+
+    const { queryByTestId } = render(
+      <AccountProfilePage params={{ address: 'incorrect-address' }} />,
+    );
+
+    expect(queryByTestId('account-profile-not-found')).toBeInTheDocument();
+    expect(queryByTestId('account-profile')).not.toBeInTheDocument();
```

Comment:
I prefer testing for content rendered by the page, rather than using test IDs. We use test IDs in cases where there is no content and no good way to test otherwise.

Reply:
what is the drawback to using test id, or this is your preference? i thought using testID is more accuracy in case the verify content may same after sometime?

Reply:
https://derekndavis.com/posts/getbytestid-overused-react-testing-library#okay-whats-so-bad-about-getbytestid